From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Mon, 8 Jan 2018 05:09:11 +0100
Subject: [PATCH] routes: Skip network restart on 'noifupdown'

Previously, setting up routes did not allow passing 'require_reboot',
so each route change would lead to a networking service restart,
rendering interface configuration options like 'noifupdown' useless.
Allow disabling network restart per-interface using the existing
'noifupdown' option.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 linux/network/interface.sls | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/linux/network/interface.sls b/linux/network/interface.sls
index 921ceac..6ebc670 100644
--- a/linux/network/interface.sls
+++ b/linux/network/interface.sls
@@ -338,6 +338,9 @@ linux_network_{{ interface_name }}_routes:
       gateway: {{ route.gateway }}
       {%- endif %}
     {%- endfor %}
+  {%- if interface.noifupdown is defined %}
+  - require_reboot: {{ interface.noifupdown }}
+  {%- endif %}

 {%- endif %}

